Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbNativeWindow.cpp

```diff
@@ -300,10 +301,120 @@ namespace AzFramework
+                m_xcbConnection,
+                m_xcbWindow,
+                screen->root,
+                0,
+                0
+                );
+            xcb_translate_coordinates_reply_t* translateReply = xcb_translate_coordinates_reply(
+                m_xcbConnection,
+                translateCookie,
+                &error
+                );
+
+            if (error || translateReply == nullptr)
+            {
+                AZ_Warning(XcbErrorWindow, error != nullptr, "Failed to translate window coordinates.");
+                free(error);
+                free(translateReply);
+                return 60;
+            }
+
+            positionX = translateReply->dst_x;
+            positionY = translateReply->dst_y;
+            free(translateReply);
+        }
+
+        xcb_randr_get_screen_resources_current_cookie_t resCookie = xcb_randr_get_screen_resources_current(m_xcbConnection, screen->root);
+        xcb_randr_get_screen_resources_current_reply_t* resReply = xcb_randr_get_screen_resources_current_reply(
+            m_xcbConnection,
+            resCookie,
+            &error);
+
+        if (error || resReply == nullptr)
+        {
+            AZ_Warning(XcbErrorWindow, error != nullptr, "Failed to get screen resources.");
+            free(error);
+            return 60;
+        }
+
+        xcb_randr_crtc_t* crtcs = xcb_randr_get_screen_resources_current_crtcs(resReply);
+        int crtcLength = xcb_randr_get_screen_resources_current_crtcs_length(resReply);
+
+        uint32_t refreshRate = 0;
+
+        for (int i = 0; i < crtcLength; i++)
+        {
+            xcb_randr_get_crtc_info_cookie_t crtcInfoCookie = xcb_randr_get_crtc_info(m_xcbConnection, crtcs[i], XCB_TIME_CURRENT_TIME);
+            xcb_randr_get_crtc_info_reply_t* crtcInfoReply = xcb_randr_get_crtc_info_reply(m_xcbConnection, crtcInfoCookie, &error);
+
+            if (error || crtcInfoReply == nullptr)
+            {
+                free(error);
+                continue;
+            }
+
+            int crtcX = crtcInfoReply->x;
+            int crtcY = crtcInfoReply->y;
+            int crtcW = crtcInfoReply->width;
+            int crtcH = crtcInfoReply->height;
+
+            if (positionX + m_width > crtcX && positionX < crtcX + crtcW &&
+                positionY + m_height > crtcY && positionY < crtcY + crtcH)
+            {
+                const uint32_t modeId = crtcInfoReply->mode;
+
+                xcb_randr_mode_info_t* modes = xcb_randr_get_screen_resources_current_modes(resReply);
+                int modesLength = xcb_randr_get_screen_resources_current_modes_length(resReply);
+
+                for (int mi = 0; mi < modesLength; mi++)
+                {
+                    auto mode = modes[mi];
+                    if (mode.id != modeId)
+                        continue;
+
+                    double refreshRatePrecise = static_cast<double>(mode.dot_clock) /
+                        (mode.htotal * mode.vtotal);
```

Comment:
Protect against divide by zero by asserting that neither `mode.htotal` nor `mode.vtotal` is zero.

Reply:
I didn't use AZ_Assert, you can see what I did in the newest commit, im not sure if its great, the warning shouldn't be possible to reach but its good to protect from divide by zero.